app/src/main/java/org/commcare/dalvik/reminders/MainActivity.kt

```diff
@@ -38,16 +39,35 @@ class MainActivity : AppCompatActivity() {
                 result.data?.let { intent ->
                     intent.getStringExtra("isPermissionGranted")?.let {permissionResult ->
                         if (permissionResult == PermissionActivity.GRANTED) {
-                            getStarted()
+                            if (hasNotificationPermission()) {
+                                getStarted()
+                            } else {
+                                requestPermissionBtn.visibility = VISIBLE
+                                var msg =  resources.getString(R.string.following_permission_not_granted)
+                                msg = msg.plus(getString(R.string.notification_permission_not_granted))
+                                statusTv.text = msg
+                                statusTv.visibility = VISIBLE
+                            }
                         } else {
                             requestPermissionBtn.visibility = VISIBLE
-                            setStatus(R.string.storage_permission_not_granted)
+
+                            var msg =  resources.getString(R.string.following_permission_not_granted)
+                            msg = msg.plus(getString(R.string.storage_permission_not_granted))
+
+                            if (!hasNotificationPermission()) {
+                                msg = msg.plus(getString(R.string.notification_permission_not_granted))
+                            }
+
+                            statusTv.text = msg
+                            statusTv.visibility = VISIBLE
+
                         }
                     }
                 }
             }
         }
 
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
```

Comment:
what's the api that need tiramisu in this method ?

Reply:
As we are using POST_NOTIFICATIONS permission  annd to suppress the lint warning it was suggested to add @RequireApi annotation

Reply:
Removed